engine/src/main/java/org/terasology/engine/logic/characters/ClientCharacterPredictionSystem.java

```diff
@@ -84,7 +84,8 @@ public void onDestroy(final BeforeDeactivateComponent event, final EntityRef ent
     @ReceiveEvent(components = {CharacterMovementComponent.class, LocationComponent.class, AliveCharacterComponent.class})
     public void onCharacterStateReceived(CharacterStateEvent state, EntityRef entity) {
         if (entity.equals(localPlayer.getCharacterEntity())) {
-            logger.trace("Received new state, sequence number: {}, buffered input size {}", state.getSequenceNumber(), inputs.size());
+            logger.atTrace().addArgument(() -> state.getSequenceNumber()).addArgument(() -> inputs.size()).
+                    log("Received new state, sequence number: {}, buffered input size {}");
```

Comment:
`getSequenceNumber()` is simple getter method and `size()` should be O(1) so impact neglectable
```suggestion
            logger.trace("Received new state, sequence number: {}, buffered input size {}", state.getSequenceNumber(), inputs.size()); //NOPMD
```